src/CodeGen_LLVM.cpp

```diff
@@ -856,18 +856,26 @@ llvm::Constant *CodeGen_LLVM::embed_metadata(const std::string &metadata_name,
         /* name */ create_string_constant(function_name)
     };
 
-    GlobalVariable *metadata = new GlobalVariable(
+    const std::string metadata_storage_name = metadata_name + "_storage";
+    GlobalVariable *metadata_storage = new GlobalVariable(
         *module,
         metadata_t_type,
         /*isConstant*/ true,
-        GlobalValue::ExternalLinkage,
+        GlobalValue::PrivateLinkage,
         ConstantStruct::get(metadata_t_type, metadata_fields),
-        metadata_name);
+        metadata_storage_name);
```

Comment:
Really minor nit: I think this is the only place metadata_storage_name is used and the name doesn't really matter, I'd just inline the definition of this name.